data_structures/arrays/max_difference.py

```diff
@@ -0,0 +1,34 @@
+def max_difference(nums: list[int]) -> int:
+    """
+    The function returns the max difference in the
+    array where result = arr[i] - arr[j] and i > j
+
+    Runtime: O(n)
+    Space: O(1)
+
+    >>> max_Difference(([2, 3, 10 , 6, 4, 8, 1])
+    8
+    >>> max_Difference(([7, 9, 5, 6, 3, 2])
+    2
+    >>> max_Difference(([10, 20, 30])
+    20
+
+    """
+    result = nums[1] - nums[0]
+
+    min_val = nums[0]
+
+    for i in range(len(nums)):
```

Comment:
Probably somethinf like that?

```python
min_value = nums[0]
max_value = nums[0]
for num in nums:
    min_value = min(min_value, item)
    max_value = max(max_value, item)

return abs(max_value - min_value)
```
Or more laconical:
```python
return max(nums) - min(nums)
```

What do you think about it?

Reply:
it would be the case if the conditions were different, it says in the file max differences between two elements nums[i] > nums[j] and i > j, so j comes before i